homeassistant/components/switch_as_x/entity.py

```diff
@@ -10,32 +10,52 @@
     SERVICE_TURN_ON,
     STATE_ON,
     STATE_UNAVAILABLE,
-    EntityCategory,
 )
-from homeassistant.core import Event, callback
-from homeassistant.helpers import entity_registry as er
-from homeassistant.helpers.entity import Entity, ToggleEntity
+from homeassistant.core import Event, HomeAssistant, callback
+from homeassistant.helpers import device_registry as dr, entity_registry as er
+from homeassistant.helpers.entity import DeviceInfo, Entity, ToggleEntity
 from homeassistant.helpers.event import async_track_state_change_event
 
 from .const import DOMAIN as SWITCH_AS_X_DOMAIN
 
 
 class BaseEntity(Entity):
-    """Represents a Switch as a X."""
+    """Represents a Switch as an X."""
 
     _attr_should_poll = False
 
     def __init__(
         self,
-        name: str,
+        hass: HomeAssistant,
+        config_entry_title: str,
         switch_entity_id: str,
         unique_id: str | None,
-        device_id: str | None,
-        entity_category: EntityCategory | None,
     ) -> None:
-        """Initialize Light Switch."""
+        """Initialize Switch as an X."""
+        registry = er.async_get(hass)
+        device_registry = dr.async_get(hass)
+        wrapped_switch = registry.async_get(switch_entity_id)
+        device_id = wrapped_switch.device_id if wrapped_switch else None
+        entity_category = wrapped_switch.entity_category if wrapped_switch else None
+        has_entity_name = wrapped_switch.has_entity_name if wrapped_switch else False
+
+        name: str | None = config_entry_title
```

Comment:
We should default to name from the other entity and not config entry title ?

Reply:
I think we should allow setting the config entry title to `None` then?

Reply:
shouldn't the entity registry entry of the wrapped entity always be the source of truth for this entity ? 

self._name = wrapped_entry.original_name
self._has_entity_name = wrapped_entry.has_entity_name

Reply:
I guess so!